#  Coursara Capstone  - Toronto Neighborhoods

## This project will compare neighborhoods in the Toronto area

By Brian Hoyler

## Part 1

create datatable with Toronto Neighborhoods

In [1]:
import pandas as pd
import numpy as nb
print('Libraries imported.')

Libraries imported.


### Install the requirements for BeautifulSoup

In [2]:
#!pip install bs4
#!pip install lxml
#!pip install html5lib
from bs4 import BeautifulSoup # this module helps in web scrapping.
import requests  # this module helps us to download a web page
print("BeautifulSoup installed")

BeautifulSoup installed


### define the URL for the data and use BeautifulSoup to scrape the data

In [14]:
#The below url contains html tables with data about Toronto Neighborhoods.
url = "https://en.wikipedia.org/wiki/List_of_neighbourhoods_in_Toronto"
url2 = "https://en.wikipedia.org/wiki/Demographics_of_Toronto_neighbourhoods"
url3 = "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"

In [15]:
# get the contents of the webpage in text format and store in a variable called data
data  = requests.get(url3).text

In [16]:
#soup = BeautifulSoup(data,"lxml")
soup = BeautifulSoup(data,"html5lib")

In [17]:
#find all html tables in the web page
tables = soup.find_all('table') # in html table is represented by the tag <table>

### The following code produces the columns for the dataframe and drops rows with unassigned boroughs

In [28]:
table_contents=[]
table=soup.find('table')
for row in table.findAll('td'):
    cell = {}
    if row.span.text=='Not assigned':
        pass
    else:
        cell['PostalCode'] = row.p.text[:3]
        cell['Borough'] = (row.span.text).split('(')[0]
        cell['Neighborhood'] = (((((row.span.text).split('(')[1]).strip(')')).replace(' /',',')).replace(')',' ')).strip(' ')
        table_contents.append(cell)
#print(table_contents)

### build dataframe and modify some of the borough names

In [29]:
df=pd.DataFrame(table_contents)
df['Borough']=df['Borough'].replace({'Downtown TorontoStn A PO Boxes25 The Esplanade':'Downtown Toronto Stn A',
                                             'East TorontoBusiness reply mail Processing Centre969 Eastern':'East Toronto Business',
                                             'EtobicokeNorthwest':'Etobicoke Northwest','East YorkEast Toronto':'East York/East Toronto',
                                             'MississaugaCanada Post Gateway Processing Centre':'Mississauga'})

## Results of Step One - Creating a dataframe of postalcodes, Boroughs and Neighborhoods

In [33]:
df

PostalCode                Borough  \
0          M3A             North York   
1          M4A             North York   
2          M5A       Downtown Toronto   
3          M6A             North York   
4          M7A           Queen's Park   
..         ...                    ...   
98         M8X              Etobicoke   
99         M4Y       Downtown Toronto   
100        M7Y  East Toronto Business   
101        M8Y              Etobicoke   
102        M8Z              Etobicoke   

                                          Neighborhood  
0                                            Parkwoods  
1                                     Victoria Village  
2                            Regent Park, Harbourfront  
3                     Lawrence Manor, Lawrence Heights  
4                        Ontario Provincial Government  
..                                                 ...  
98       The Kingsway, Montgomery Road, Old Mill North  
99                                Church and Wellesley  
100                                     Enclave of M4L  
101  Old Mill South, King's Mill Park, Sunnylea, Hu...  
102  Mimico NW, The Queensway West, South of Bloor,...  

[103 rows x 3 columns]

In [49]:
df.shape

(103, 3)

## First portion is complete.  The dataframe has 103 rows.